cpp/src/arrow/flight/sql/server.h

```diff
@@ -225,6 +225,27 @@ struct ARROW_FLIGHT_SQL_EXPORT ActionCreatePreparedStatementResult {
   std::string prepared_statement_handle;
 };
 
+/// \brief A request to close the open client session.
+struct ARROW_FLIGHT_SQL_EXPORT ActionCloseSessionRequest {};
```

Comment:
Ditto here - how is the server supposed to get the session identifier?

Reply:
[see [above](https://github.com/apache/arrow/pull/34817#pullrequestreview-1366808831) for discussion]

More specifically, the SetSessionOptions implementation would inject the Set-Cookie response header as appropriate.  Or in the alternate case of an existing session, it would be accessed via the request headers.